packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1401,37 +1313,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     public createSummary(): ISummaryTree {
         const builder = new SummaryTreeBuilder();
 
-        // Attaching graph of some stores can cause other stores to get bound too.
-        // So keep taking summary until no new stores get bound.
-        let notBoundContextsLength: number;
-        do {
-            const builderTree = builder.summary.tree;
-            notBoundContextsLength = this.dataStores.contexts.notBoundLength();
-            // Iterate over each data store and ask it to snapshot
-            Array.from(this.dataStores.contexts)
-                .filter(([key, _]) =>
-                    // Take summary of bounded data stores only, make sure we haven't summarized them already
-                    // and no attach op has been fired for that data store because for loader versions <= 0.24
-                    // we set attach state as "attaching" before taking createNew summary.
-                    !(this.dataStores.contexts.isNotBound(key)
-                        || builderTree[key]
-                        || this.dataStores.attachOpFiredForDataStore.has(key)),
-                )
-                .map(([key, value]) => {
-                    let dataStoreSummary: ISummarizeResult;
-                    if (value.isLoaded) {
-                        const snapshot = value.generateAttachMessage().snapshot;
-                        dataStoreSummary = convertToSummaryTree(snapshot, true);
-                    } else {
-                        // If this data store is not yet loaded, then there should be no changes in the snapshot from
-                        // which it was created as it is detached container. So just use the previous snapshot.
-                        assert(!!this.context.baseSnapshot,
-                            "BaseSnapshot should be there as detached container loaded from snapshot");
-                        dataStoreSummary = convertSnapshotTreeToSummaryTree(this.context.baseSnapshot.trees[key]);
-                    }
-                    builder.addWithStats(key, dataStoreSummary);
-                });
-        } while (notBoundContextsLength !== this.dataStores.contexts.notBoundLength());
+        this.dataStores.createSummary(builder);
```

Comment:
Similar to above comment. createSummary should return ISummaryTree and SummaryTreeBuilder should move to dataStores.

Reply:
again, this is how blobs works too. i don't like the pattern, but not looking to introduce a new on here

Reply:
Sounds reasonable.

Reply:
Also, since this is all in same package, I'm not too worried about it- we can always revisit if it seems problematic.

Reply:
for now yes. Longer term i would love to see datastore in its own package, more like a dds.